Ts verify

bigtrak opened the pull request:
These changes ... do not change anything in the testsuite.  It is exactly the same, test-wise,
as it was before.

What they do is make the generation of the platform dirs to be completely automated,
and increase uniformity between the platforms ancillary files.  You can 'rm -rf' a platform
directory, and .. on a platform where gprolog works ... it will re-create it exactly.

Although tuples.gz has changed, 'tuples' itself is identical to the previous version, as are the generated
files.   Test results are identical to previous.  The build process has changed to make the compressed
tuples.gz files the same as the prior ... if changes did not occur to the test specification.

Makefile.genspec is reformatted to take better advantage of VCS facilities, to ensure changes are
easily spotted.

The deletions are files that are no longer generated or used by the testsuite.

I've also added some notes, so that someone else who has never created the platform from the specification, has some guidelines and hints of how to proceed.

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
 
+## The test specification is in
+##	src/specification/spec-grouped.pl
+##	src/specification/spec-ungrouped.pl		-- no longer works
```
If that file doesn't work, let's just remove it.

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
+##	allow for grouping of tests, which speeds test harness
+##	execution greatly by reducing overhead ... when it works.
 
+## Recommended method to generate a new specification, is to:
```
"specification" here means a different variant on an existing test or a new platform?

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
+## 1) move the existing platform configurations to a new subdir
+## 2) gunzip -c tuples.gz >tuples   for each OLD platform in that subdir.
+## 3) generate the new specification
+##		make -f Makefile.gespec gen-clean gen-all
```
sp: Makefile.genspec

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
+## 2) gunzip -c tuples.gz >tuples   for each OLD platform in that subdir.
+## 3) generate the new specification
+##		make -f Makefile.gespec gen-clean gen-all
+##	nB: You can make 1 platform by specifying it as a target;
```
Could you clarify this a bit?

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
+##	nB: You can make 1 platform by specifying it as a target;
+##	when testing to see if anything works at all in the generator,
+##	that is a good place to start!
+## 4) Use both diff and git to compare the results.
```
What is being compared here? The generated CMakeLists.txt file?

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
+## is to re-generate the existing specification, using the above process.
+## 1) If you get byte-for-byte results, everything is OK
+## 2) If not, there is likely a prolog issue,
+##    and you need to find a platform which generates byte-for-byte identical
```
Gross. That prolog stuff has to go.

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
+## BLANK ABOVE
+
+DEPRECATED_SUBDIRS= \
+	arm64-unknown-linux		\
```
The only SUBDIRs (i.e., platforms) not deprecated are x86_64-unknown-linux2.4, ppc64_linux, aarch64-unknown-linux. All the others are dead- for testing purposes. We may someday get back around to testing on FreeBSD.

hainest commented on Makefile.genspec:
```diff
@@ -1,9 +1,138 @@
+## The "ungrouped" spec seems to be historical;, and as of 2020 doesn't work.
+##	it is left here for consideration, and to not lose info.
+
+SPEC=spec-ungrouped.pl
```
Why the extra SPEC?

hainest commented on Makefile.genspec:
```diff
@@ -14,23 +143,80 @@ usage:
+
+## XXX could use filter/filter-out, or put this in a script, leave it for now.
+
+$(SUBDIRS:%=%/.gitignore): Makefile.genspec
```
Splitting these by platform is overkill. I would just make a "master" .gitignore and copy it into the new dirs.

hainest commented:
This is great! I just have a few items I don't fully understand; see the review comments.

bigtrak commented:
Tim, this PR is setup as a checkins/ pull request to 
make a "new baseline" of -exactly-as-before- performance and
reliability.  The files are the same, but now they are generated
by a repeatable mechanism.

Once this setup is in the mainline, in the future I could then make
some of the other changes you propose, which may change or remove
functionality or alter results of the testsuite.


I have addressed some of the issues you mention which follow that model
shortly, as well as correct spelling typo.

I've also reworded the notes, and adjusted them after determining
that the ubuntu-provided gprolog was the source of failure, not
gprolog itself.    gprolog works both pie and non-pie on those
systems, by the way, that wasn't the issue, just the ubuntu packaging.

I contacted Matt Legendre to ask about the status of spec-ungrouped.pl,
he thinks it can go the way of the dodo bird.   As that will not cause
changes in the test results, that has been taken care of as well.

I think the PR should still valid, as it should follow the branch head,
even after the additional commits.

Bolo

hainest commented:
LGTM. You can merge whenever you like.
